manage discovery plugin config file

jhoblitt opened the pull request:
In our production environment, `/etc/foreman-proxy/settings.d` is purged of un-managed files.  This results in `/etc/foreman-proxy/settings.d/discovery.yml` being deleted and the discovery plugin becoming disabled.

ekohl reviewed (commented):
It really surprises me we never managed this in the past. This should probably use [plugin::module](https://github.com/theforeman/puppet-foreman_proxy/blob/master/manifests/plugin/module.pp)

jhoblitt commented:
@ekohl Doh! I completely missed `foreman_proxy::plugin::module`.  I will update the PR...

jhoblitt commented:
I have refactored `foreman_proxy::plugin::discovery` to use `foreman_proxy::plugin::module`.  I confess to attempting to be lazy about testing the new parameters.  I also added a basic acceptance test.

ekohl reviewed (approved):
Thanks!

ekohl commented on manifests/plugin/discovery.pp:
```diff
@@ -15,16 +20,21 @@
-  foreman_proxy::plugin { 'discovery':
+  foreman_proxy::plugin::module { 'discovery':
+    enabled   => $enabled,
+    feature   => 'Discovery',
```
Technically redundant, but doesn't hurt.

jhoblitt replied:
I would be happy to remove that... I was copying the style of other classes under the same path.

jhoblitt replied:
https://github.com/theforeman/puppet-foreman_proxy/pull/773

ekohl commented:
And of course this is submitted right after I did 2 releases today (because I messed one up), though I'd probably want to get this into 3.4.
